### PR TITLE
Streamline timm integration

### PR DESCRIPTION
**Pull request by vuptran**

### CODE REVIEW

File: models/sesemi.py [hunk 1]
```diff
@@ -14,31 +14,48 @@
...
+    # pretrained ImageNet weights. For a summary of their ImageNet performance, see
+    # <https://github.com/rwightman/pytorch-image-models/blob/master/results/results-imagenet.csv>.
 
-    # Standard EfficientNet models.
+    # Residual models and variants.
+    # Compared with the defaults, the "d" variants (e.g., resnet50d, resnest50d)
+    # replace the 7x7 conv in the input stem with three 3x3 convs.
+    # And in the downsampling block, a 2x2 avg_pool with stride 2 is added before conv,
+    # whose stride is changed to 1.
+    # Described in `Bag of Tricks <https://arxiv.org/abs/1812.01187>`.
```
Comment (tigist-d) [inline models/sesemi.py]: There looks to be a new set of ResNet models supported by timm based on the following paper:
[Revisiting ResNets: Improved Training and Scaling Strategies](https://arxiv.org/abs/2103.07579)

They are listed as:
resnetrs50, resnetrs101, resnetrs152, resnetrs200, resnetrs270, resnetrs350, resnetrs420

Would these be worthwhile including here?

File: models/timm.py [hunk 2]
```diff
@@ -15,26 +15,20 @@
...
         self.encoder = torch.hub.load(
             PYTORCH_IMAGE_MODELS_REPO,
             backbone, 
-            pretrained=pretrained,
+            pretrained,
         )
-        self.in_features = self.encoder.classifier.in_features
+        for m in self.encoder.modules():
+            if isinstance(m, (nn.Linear)):
+                self.in_features = m.in_features
```
Comment (tigist-d) [inline models/timm.py]: I believe that all timm models support the following syntax to get the number of features produced by the models:

```
self.in_features = self.encoder.num_features
```

File: models/sesemi.py (hunk 1 above)
Comment (vuptran) [reply models/sesemi.py]: Up to resnetrs200, yes. The others are too big for their size with little accuracy gains.

**Review by tigist-d — APPROVED** [review]
